crates/cdk/src/publish.rs

```diff
@@ -101,11 +115,7 @@ impl PublishCmd {
 
             // --push only, needs ipkg file
             (false, true) => {
-                let pkgfile = &self
-                    .package_meta
```

Comment:
We should put the logic from lines 68-72 here and remove `unreachable`:
```
            // --push only, needs ipkg file
            (false, true) => {
                let pkgfile = &self
                    .ipkg_file
                    .as_ref()
                    .ok_or_else(|| anyhow::anyhow!("need to specify --ipkg-file when using --push"))?;
                package_push(self, pkgfile, &access)?;
            }
```

We should also make `ipkg_file` required if  `push` is specified using clap annotations.

Reply:
we should decide on how to organize this and apply it to both smdk and cdk. Currently cdk has `no_build` and smdk doesn't have it as well

Reply:
It is probably the time to re-organize the whole function. Split it into separate functions for better readability.

Because the current change with one more if and `unreachable` macro doesn't look good.

Reply:
I refactored using functions but don't thing there is a way to specify to require an arg if one flag is true and other is false in clap

Reply:
created issue about `no-build` flag https://github.com/infinyon/fluvio/issues/3371